File: 3189-find-champion-ii/3189-find-champion-ii.js
/**
 * @param {number} n
 * @param {number[][]} edges
 * @return {number}
 */
var findChampion = function(n, edges) {
    const indegree = new Array(n).fill(0);
    for(const e of edges) {
        indegree[e[1]]++;
    }

    let ans = [];
    for(let i = 0; i<n; i++) {
        if(indegree[i] === 0) ans.push(i);
    }

    return ans.length == 1 ? ans[0] : -1;
};

File: 3189-find-champion-ii/3189-find-champion-ii.java
class Solution {
    public int findChampion(int n, int[][] edges) {
        int[] indegree = new int[n];
        for(int[] e : edges) {
            indegree[e[1]]++;
        }

        List<Integer> ans = new ArrayList<>();
        for(int i = 0; i<n; i++) {
            if(indegree[i] == 0) ans.add(i);
        }

        return ans.size() == 1 ? ans.get(0) : -1;
    }
}